forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js

```diff
@@ -0,0 +1,621 @@
+import React, { useState } from "react";
+import { useSelector } from "react-redux";
+import clsx from "clsx";
+import { makeStyles } from "@material-ui/core/styles";
+import Drawer from "@material-ui/core/Drawer";
+import Button from "@material-ui/core/Button";
+import List from "@material-ui/core/List";
+import Divider from "@material-ui/core/Divider";
+import "../../ServiceFlow.scss";
+import { saveFilters } from "../../../../apiManager/services/bpmTaskServices";
+
+const useStyles = makeStyles({
+  list: {
+    width: 380,
+  },
+  fullList: {
+    width: "auto",
+  },
+});
+
+export default function CreateNewFilterDrawer() {
+  const classes = useStyles();
+  const [state, setState] = React.useState({
+    top: false,
+    left: false,
+    bottom: false,
+    right: false,
+  });
+  const [filterName, setFilterName] = useState("");
+  const [showUndefinedVaribale, setShowUndefinedVaribale] = useState(false);
+  const [inputVisibility, setInputVisibility] = useState({});
+  const [definitionKeyId, setDefinitionKeyId] = useState("");
+  const [candidateGroup, setCandidateGroup] = useState([]);
+  const [taskName, setTaskName] = useState([]);
+  const [includeAssignedTasks, setIncludeAssignedTasks] = useState(false);
+  const [rows, setRows] = useState([]);
+  const [newRow, setNewRow] = useState({ variableName: "", labelName: "" });
+  const [permissions, setPermissions] = useState("");
+  const [identifierId, setIdentifierId] = useState("");
+  const [selectUserGroupIcon, setSelectUserGroupIcon] = useState("");
+  const [specificUserGroup, setSpecificUserGroup] = useState("");
+  const userName = useSelector((state) => state.user?.userDetail?.name);
+
+  //Function for collecting the input data
+  const handleSubmit = () => {
+    const data = {
+      name: filterName,
+      criteria: {
+        processDefinitionKey: definitionKeyId,
+        candidateGroup: candidateGroup,
+        assignee: taskName,
+        includeAssignedTasks: false,
+      },
+      variables: [
+        ...rows.map((row) => ({
+          name: row.variableName,
+          label: row.labelName,
+        })),
+        {
+          name: newRow.variableName,
+          label: newRow.labelName,
+        },
+      ],
+      users: [userName],
+      roles: [selectUserGroupIcon],
+      identifierId: identifierId,
+    };
+    saveFilters(data)
+      .then(toggleDrawer("left", false))
```

Comment:
toggleDrawer remove this function and you can directly call the open and close drawer state